### PR TITLE
Small fixes driven by integration test development

### PR DESCRIPTION
**Pull request by bemasc**
Includes:
- Report a specific SOCKS error for DNS resolution failure
- Fix logic/scoping error in churn-pipe, discovered during IPv6 testing

Fixes https://github.com/uProxy/uproxy/issues/794

[Review on Reviewable](https://reviewable.io/reviews/uproxy/uproxy-lib/217)

### CODE REVIEW

File: src/integration-tests/socks-echo/base-spec.core-env.ts [hunk 1]
```diff
@@ -237,11 +237,11 @@ export function socksEchoTestDescription(useChurn:boolean) {
...
       // corporate DNS can drop responses that resolve to local network
       // addresses.  Accordingly, the error code may either indicate
-      // a generic failure (if resolution fails) or NOT_ALLOWED if name
+      // HOST_UNREACHABLE (if resolution fails) or NOT_ALLOWED if name
       // resolution succeeds.  However, to avoid portscanning leaks
-      // (https://github.com/uProxy/uproxy/issues/809) both will be reported
+      // (https://github.com/uProxy/uproxy/issues/809) NOT_ALLOWED will be reported
       // as FAILURE
-      expect(e.reply).toEqual(socks.Reply.FAILURE);
+      expect([socks.Reply.HOST_UNREACHABLE, socks.Reply.FAILURE]).toContain(e.reply);
```
Comment (trevj) [inline src/integration-tests/socks-echo/base-spec.core-env.ts]: I think it's a lot more readable the other way around!

Comment (bemasc) [inline src/integration-tests/socks-echo/base-spec.core-env.ts]: Yeah, but I don't think jasmine lets you say `expect(a).toEqual(b).or(c)`.

File: src/integration-tests/socks-echo/base-spec.core-env.ts [hunk 2]
```diff
@@ -332,10 +332,7 @@ export function socksEchoTestDescription(useChurn:boolean) {
     }).then(done);
   });
 
-  // Disabled because HOST_UNREACHABLE is not yet exposed in freedom-for-chrome's
-  // implementation of the core.tcpsocket API.
-  //  https://github.com/freedomjs/freedom-for-chrome/issues/73
-  xit('attempt to connect to a nonexistent DNS name', (done) => {
+  it('attempt to connect to a nonexistent DNS name', (done) => {
```
Comment (trevj) [inline src/integration-tests/socks-echo/base-spec.core-env.ts]: Oh! Does this fix the warning we currently get on the console, too?

**Comment by trevj** [comment]
Nice!

:+1:

File: src/integration-tests/socks-echo/base-spec.core-env.ts (hunk 2 above)
Comment (bemasc) [inline src/integration-tests/socks-echo/base-spec.core-env.ts]: I wouldn't expect it to help.  That sounds like something that could be caused by a missing error check in freedom-for-chrome's core.tcpsocket.

**Pull request merged**